packages/rest/src/session/CredsForSessCfg.ts

```diff
@@ -0,0 +1,207 @@
+     *      );
+     *      mySession = new Session(sessCfgWithCreds);
+     *
+     * @returns A session configuration object with credentials added
+     *          to the initialSessCfg. Its intended use is for our
+     *          caller to create a session for a REST Client.
+     */
+    public static async addCredsOrPrompt<T>(
+        initialSessCfg: T,
+        cmdArgs: ICommandArguments,
+        options: IOptionsForAddCreds = {}
+    ): Promise<T> {
+        const impLogger = Logger.getImperativeLogger();
+
+        const optionDefaults: IOptionsForAddCreds = {
+            requestToken: false,
+            doPrompting: true
+        };
+
+        // override our defaults with what our caller wants.
+        const optsToUse = {...optionDefaults, ...options};
+
+        // initialize session config object
+        const finalSessCfg: any = initialSessCfg;
+
+        // confirm which credential properties were supplied by user
+        let tokenValExists = false;
+        let tokenTypeExists = false;
+        let userExists = false;
+        let passExists = false;
+        if (cmdArgs.tokenValue !== undefined &&
+            cmdArgs.tokenValue !== null &&
+            cmdArgs.tokenValue.length > 0)
+        {
+            tokenValExists = true;
+        }
+        if (cmdArgs.tokenType !== undefined &&
+            cmdArgs.tokenType !== null &&
+            cmdArgs.tokenType.length > 0)
+        {
+            tokenTypeExists = true;
+        }
+        if (cmdArgs.user !== undefined &&
+            cmdArgs.user !== null &&
+            cmdArgs.user.length > 0)
+        {
+            userExists = true;
+        }
+        if (cmdArgs.password !== undefined &&
+            cmdArgs.password !== null &&
+            cmdArgs.password.length > 0)
+        {
+            passExists = true;
+        }
+
+        // set credential values from cmdLine (or environment, or profile)
+        if (optsToUse.requestToken) {
+            // ignoring tokenValue, will ensure that user & password are used for authentication
+            tokenValExists = false;
+        }
+
+        if (userExists || tokenValExists === false) {
+            /* When a user name is supplied, that is what we use.
+             * If neither user name or token is supplied,
+             * we prompt for username and password, as needed.
+             */
+            if (optsToUse.requestToken) {
+                // Set our type to token to get a token from user and pass
+                impLogger.debug("Using basic authentication to get token");
+                finalSessCfg.type = SessConstants.AUTH_TYPE_TOKEN;
+                if (tokenTypeExists) {
+                    finalSessCfg.tokenType = cmdArgs.tokenType;
+                } else {
+                    // When no tokenType supplied, user wants bearer
+                    finalSessCfg.tokenType = SessConstants.TOKEN_TYPE_LTPA;  // TODO:Gene: change to TOKEN_TYPE_APIML
```

Comment:
Should this TODO be removed?

Reply:
Similar TODOs in tests

Reply:
The login command currently in the CLI generates zosmf tokens. If we change the default token value to APIML now, then we will try to use a zosmf token with an APIML token type. These TODO items are reminders that we must change these values once we move the APIML token logic from the POC plugin into the CLI.